### PR TITLE
LPS-193361 - Add support for system relationships for system objects

### PR DESCRIPTION
**Pull request by G-Sa**
related to https://liferay.atlassian.net/browse/LPS-193361

### CODE REVIEW

**Comment by G-Sa** [comment]
ci:test:sf

**Comment by G-Sa** [comment]
ci:test:object

**Comment by G-Sa** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 1]
```diff
@@ -882,9 +903,14 @@ private ObjectRelationship _addObjectRelationship(
 			long userId, long objectDefinitionId1, long objectDefinitionId2,
 			long parameterObjectFieldId, String deletionType,
 			Map<Locale, String> labelMap, String name, boolean reverse,
-			String type)
+			boolean system, String type)
 		throws PortalException {
 
+		if (system && !SystemUtil.allowManageSystemEntities()) {
+			throw new ObjectRelationshipSystemException(
+				"Only allowed bundles can create system relationships");
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:911]: I think this word is missing: "Only allowed bundles can create system **object** relationships"

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 2]
```diff
@@ -882,9 +903,14 @@ private ObjectRelationship _addObjectRelationship(
 			long userId, long objectDefinitionId1, long objectDefinitionId2,
 			long parameterObjectFieldId, String deletionType,
 			Map<Locale, String> labelMap, String name, boolean reverse,
-			String type)
+			boolean system, String type)
 		throws PortalException {
 
+		if (system && !SystemUtil.allowManageSystemEntities()) {
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:909]: Question to Product Team: Is adding a system object relationship from any object definition possible? Or is it going to be only allowed to modifiable system object definition?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java (hunk 2 above)
Comment (marcelabc) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:909]: Answer: Only Modifiable System Object Definition can create system object relationships

Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:909]: I believe the condition should be:
```
if (system && objectDefinition.isModifiable() && objectDefinition.isSystem() && 
!SystemObjectDefinitionManagementChecker.isInvokerBundleAllowed()) {}
```

**Review by marcelabc — CHANGES_REQUESTED** [review]
Adding more comments to the previous review :+1:

**Comment by marcelabc** [comment]
We must remember to adapt the logic from `_addObjectDefinitionResources()` in `ObjectDefinitionResourceImpl` class;

**Pull request closed without merging**